spring-rabbit/src/test/java/org/springframework/amqp/rabbit/connection/CachingConnectionFactoryIntegrationTests.java

```diff
@@ -255,14 +256,16 @@ public void testSendAndReceiveFromVolatileQueue() throws Exception {
 
 	@Test
 	public void testReceiveFromNonExistentVirtualHost() throws Exception {
-
 		connectionFactory.setVirtualHost("non-existent");
 		RabbitTemplate template = new RabbitTemplate(connectionFactory);
 		// Wrong vhost is very unfriendly to client - the exception has no clue (just an EOF)
-		exception.expect(AmqpIOException.class);
-		String result = (String) template.receiveAndConvert("foo");
-		assertEquals("message", result);
-
+		try {
+			template.receiveAndConvert("foo");
+			fail("Exception expected");
+		}
+		catch (AmqpIOException | AmqpAuthenticationException e) {
+			// expected
```

Comment:
I understand that it doesn't have much sense, but seems for me we should finish this test with some assertion. 
```
exception.expect(Matchers.anyOf(Matchers.instanceOf(AmqpIOException.class),
				Matchers.instanceOf(AmqpAuthenticationException.class)));
```
?

Just let me know and I'll fix on merge!